src/modules/map/components/openlayers/utils/useVectorLayer.composable.js

```diff
@@ -23,7 +23,8 @@ export default function useVectorLayer(
     features,
     zIndex = -1,
     styleFunction = highlightFeatureStyle,
-    onFeatureSelectCallback = () => {}
+    onFeatureSelectCallback = () => {},
+    deselectAfterSelect = false // If true, the selected feature will be deselected after the callback is called
```

Comment:
could you change the way params are given and use a config object approach?
this starts to have too many arguments to be able to know which is what